modules/document_repository/php/editfile.class.inc

```diff
@@ -0,0 +1,183 @@
+<?php
+/**
+ * It includes two functions.
+ * Post method handles updating a file.
+ * Delete method handles deleting a file.
```

Comment:
I don't really understand why this isn't just `/document_repository/files/$filename` with GET, PUT, and DELETE methods. Separating the endpoints for editing/deleting from retrieving seems arbitrary.